go/adbc/driver/databricks/cloudfetch_e2e_test.go

```diff
@@ -0,0 +1,496 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build integration
+// +build integration
+
+package databricks_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/apache/arrow-adbc/go/adbc/driver/databricks"
+	"github.com/apache/arrow-go/v18/arrow/memory"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestCloudFetchE2E_BasicConnection tests basic connectivity
+// CloudFetch is handled automatically by the databricks-sql-go driver
+func TestCloudFetchE2E_BasicConnection(t *testing.T) {
+	host, token, httpPath, catalog, schema := getDatabricksConfig(t)
+
+	driver := databricks.NewDriver(memory.DefaultAllocator)
+
+	opts := map[string]string{
+		databricks.OptionServerHostname: host,
+		databricks.OptionHTTPPath:       httpPath,
+		databricks.OptionAccessToken:    token,
+		databricks.OptionCatalog:        catalog,
+		databricks.OptionSchema:         schema,
+	}
+
+	db, err := driver.NewDatabase(opts)
+	require.NoError(t, err)
+	defer func() { _ = db.Close() }()
```

Comment:
you should use `validation.CheckedClose` for this instead of just dropping/ignoring the result